src/frontend/components/ProductPrice/ProductPrice.tsx

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import getSymbolFromCurrency from 'currency-symbol-map';
+import { Money } from '../../protos/demo';
+import { useCurrency } from '../../providers/Currency.provider';
+
+interface IProps {
+  price: Money;
+}
+
+const ProductPrice = ({ price }: IProps) => {
+  const [{ units, currencyCode }, setParsedPrice] = useState<Money>(price);
+  const { convert, selectedCurrency } = useCurrency();
+
+  const convertPrice = useCallback(async () => {
+    const result = await convert(price);
+
+    setParsedPrice(result);
+  }, [convert, price]);
+
+  const currencySymbol = useMemo(
+    () => getSymbolFromCurrency(currencyCode) || selectedCurrency,
+    [currencyCode, selectedCurrency]
+  );
+
+  useEffect(() => {
+    convertPrice();
+  }, [selectedCurrency, price]);
+
+  return (
+    <span>
+      {currencySymbol}{units}
```

Comment:
> Ah, btw... The loadgen is failing because the FE url is different now. Maybe we could consider either moving from `8081` to `8080` or updating the loadgen endpoint as well.

I fixed the endpoint for the loadgen but the scenarios are no longer "valid" as the app does not support the same POST/GET endpoints. I'll update it with better next as part of a follow up PR

Reply:
Fixed